mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -255,18 +256,37 @@ func (sp *StateProcessor[
 		)
 	}
 
+	// Ensure the EVM inflation withdrawal is the first withdrawal.
+	if numWithdrawals == 0 {
+		return ErrNoWithdrawals
+	}
```

Comment:
is this check needed? We build ExpectedWithdrawals with at least 1 withdrawal and we check the length above.
Consider dropping it